README.md

```diff
@@ -9,3 +9,681 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+
+```ts
+new RouterMetaBuilder()
+  .path('/pets/')
+  .allowAnonymous()
+  .post(({ body }) => {
+
+  })
+```
+
+What ever is posted will be in body... but maybe I want to whitelist what makes it to my handler.
+
+```ts
+@autoinject
+class postPetMediaType {
+  constructor(private petMediaType: PetMediaType) {}
+  mediaType: string = 'application/pet+json';
+  formatForResponse(pets: IPet[]): any {
+    throw new Error('not supported');
+  }
+  formatFromRequest(body: any): any {
+    const { firstName, lastName } = body;
+    return { firstName, lastName };
+  }
+}
+// and in my router...
+ new RouterMetaBuilder()
+    .path('/pets/')
+    .allowAnonymous()
+    .mediaType(this.petsMediaType)
+    .mediaType(this.postPetMediaType)
+    .post(({ model }) => {
+
+    })
+```
+
+Now, `model` is what we'd expect.
+
+> The MediaType architecture could use a v2 to embrace that it is really an adapter for the handler (which use a media type), it is not a media type!
+
+Another example is when we need to make a breaking change...
+
+```ts
+class OldPetMediaType {
+  mediaType: string = 'application/json';
+  formatForResponse({ firstName, lastName }: IPet): any {
+    return { name: `${firstName} ${lastName}` };
+  }
+  formatFromRequest(body: any): any {
+    const { name } = body;
+    return { firstName: name, lastName: '' };
+  }
+}
+
+// tslint:disable-next-line: max-classes-per-file
+class PetMediaType {
+  mediaType: string = 'application/pet+json';
+  formatForResponse({ firstName, lastName }: IPet): any {
+    return { firstName, lastName };
+  }
+  formatFromRequest(body: any): any {
+    const { firstName, lastName } = body;
+    return { firstName, lastName };
+  }
+}
+```
+
+Now I can support the new data structure without breaking the old one, and encourage consumers of the old one to switch. I can then remove very little code, as this adapting happened at the right layer insulating most of my code from it.
+
+> Of note, the second parameter to the format methods is `req` in the rare case you need something raw from the req, eg a query param that is doing column filtering.
+
+## What if I need a different handler based on the media type?
+TODO
+
+## What if I need to lock down most of my endpoints?
+So far, we haven't done any authorization, lets lock it down.
+
+If we remove allow anonymous, we will start getting a 401
+```ts
+new RouterMetaBuilder()
+  .path('/pets/')
+  .mediaType(this.petsMediaType)
+  .get(() => {
+    return this.pets;
+  }),
+```
+
+```sh
+curl -H 'accept:application/pets+json'  http://localhost:4000/pets/
+{"status":401,"message":"Authorization is required"}
+```
+
+Security is going to be different for every application, so we provide a hook for you to bring your own security.
+
+
+```ts
+
+import 'reflect-metadata'; // Required by aurelia-dependency-injection
+
+import { ISecurityContext, ISecurityContextProvider, RouteProvider, SecurityContextProvider } from '@symbiotic/express-router-adapter';
```

Comment:
break imports onto multiple line sso it doesnt clip

![image](https://user-images.githubusercontent.com/1964631/61544939-61e55c80-aa14-11e9-9f6c-456cd8ccae53.png)